synchronizer/batches.go

```diff
@@ -440,3 +456,52 @@ func (bs *BatchSynchronizer) resolveWithMember(
 		BatchNum: batch.Number,
 	}, nil
 }
+
+func (bs *BatchSynchronizer) startOffchainDataGapsDetection(ctx context.Context) {
+	log.Info("starting handling unresolved batches")
+	for {
+		delay := time.NewTimer(time.Minute)
+		select {
+		case <-delay.C:
+			if err := bs.detectOffchainDataGaps(ctx); err != nil {
+				log.Error(err)
+			}
+		case <-bs.stop:
+			return
+		}
+	}
+}
+
+// detectOffchainDataGaps detects offchain data gaps and reports them in logs and the service state.
+func (bs *BatchSynchronizer) detectOffchainDataGaps(ctx context.Context) error {
+	// Detect offchain data gaps
+	gaps, err := detectOffchainDataGaps(ctx, bs.db)
+	if err != nil {
+		return fmt.Errorf("failed to detect offchain data gaps: %v", err)
+	}
+
+	// No gaps found, all good
+	if len(gaps) == 0 {
+		return nil
+	}
+
+	// Log the detected gaps
+	gapsRaw := new(bytes.Buffer)
+	for key, value := range gaps {
```

Comment:
Should we add these key values to the `bs.offchainDataGaps` here, so we do not iterate twice through the `gaps` map.

Reply:
Done